test/fixtures/when.go

```diff
@@ -42,6 +42,26 @@ type When struct {
 	portForwarderStopChannels map[string]chan struct{}
 }
 
+// SendMessageTo sends msg to one of the pods in http source vertex.
+func (w *When) SendMessageTo(pipelineName string, vertexName string, msg []byte) *When {
+	w.t.Helper()
+	labelSelector := fmt.Sprintf("%s=%s,%s=%s", dfv1.KeyPipelineName, pipelineName, dfv1.KeyVertexName, vertexName)
+	ctx := context.Background()
+	podList, err := w.kubeClient.CoreV1().Pods(Namespace).List(ctx, metav1.ListOptions{LabelSelector: labelSelector, FieldSelector: "status.phase=Running"})
```

Comment:
Should we cache or something the result of `Pod().List()` since this function will be called for every `SendMessageTo()`? Won't it be too many calls to kube-api server?

Reply:
That's a very good point, will add in-memory cache. +1